sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -199,7 +199,7 @@ case class FileSourceScanExec(
     ret
   }
 
-  override val (outputPartitioning, outputOrdering): (Partitioning, Seq[SortOrder]) = {
+  override lazy val (outputPartitioning, outputOrdering): (Partitioning, Seq[SortOrder]) = {
```

Comment:
That happens on the driver so no need for the `lazy` here.

Reply:
it'd be computed anyway, though, when we create a new `FileSourceScanExec` in the canonicalization process, if it is not lazy, so I'd say that this is needed, as well as all the others.